webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala

```diff
@@ -1456,23 +1458,23 @@ object RuleExpectedReportBuilder extends Loggable {
     val getTrackingVariableCardinality : (Seq[String], Seq[String]) = {
       val boundingVar = vars.trackerVariable.spec.boundingVariable.getOrElse(vars.trackerVariable.spec.name)
       // now the cardinality is the length of the boundingVariable
-      (vars.expandedVars.get(boundingVar), vars.originalVars.get(boundingVar)) match {
-        case (None, None) =>
+      (vars.expandedVars.filter(_._1._2 == boundingVar), vars.originalVars.filter(_._1._2 == boundingVar)) match {
```

Comment:
i'm unsure of the meaning of this change: techniques from Technique Editor dont have the trackingvariable  (it is bound to itself), and old technique don't have blocks.

Reply:
this is the same logic, apart that I can't use get as i just want to use the value of the variable, no need the patch